### ISSUE TITLE
reading a series of empty cells returns a 0x0 tibble, rather than an MxN table of NA values

### ISSUE THREAD

**Issue by mkoohafkan** [tidyverse/readxl#550]
Consider reading a blank Excel spreadsheet:
```
read_excel("blank.xlsx", "Sheet1", "A1:C1", na = "")
# A tibble: 0 x 0
```
I would expect this to instead return a 1x3 tibble of `NA` values.

**Comment by mkoohafkan** [comment]
This issue also causes inconsistent handling of NA values, i.e. the behavior of `read_excel(na = "")` is different from `read_excel(na = "NA")` or even `read_excel(na = " ")`
```
# cells are not blank, contain a single whitespace
read_excel("spaces.xlsx", "Sheet1", "A1:C1", na = " ", col_names = FALSE)
# A tibble: 1 x 3
  ...1  ...2  ...3 
  <lgl> <lgl> <lgl>
1 NA    NA    NA 
```
I think the return value should be consistent regardless of the value of the `na` argument.

**Comment by jennybc** [comment]
I don't understand this most recent example and this statement:

> I think the return value should be consistent regardless of the value of the na argument.

In this example, we read 3 non-empty cells and in each case the contents match the string for `na`. Therefore they all come in as `NA`.

**Comment by mkoohafkan** [comment]
Exactly. But in the first example we read in empty cells where the `NA` match is `""` (empty) so I would expect to get NA values back in that case as well.

**Comment by jennybc** [comment]
Oh yeah, my instinct is that the original example is unexpected. But not the second one.

**Comment by mkoohafkan** [comment]
Agreed, but I wasn't sure what the preferred convention was or if there was a specific reason for the result of the first test case. More generally, my initial expectation (and the majority of my experience using the package so far) is that `read_excel` returns a tibble with dimensions equivalent to the specified cell range, regardless of content.

**Comment by KHBHH** [comment]
Hi,

 is this issue solved?

**Comment by batpigandme** [comment]
Hi @KHBHH,
We use [GitHub's keywords](https://help.github.com/en/articles/closing-issues-using-keywords) in commits/pull requests to close issues when we fix them. So, the issue will be closed (and the related PR linked) when it's been solved.

**Comment by KHBHH** [comment]
Hi,

Thank you for the notice.

I am reading an excel file that contains some blank cells indicating they are unkwon values. I want to replace these blank cells by **NA** string. I am using **read_excel** function, but when I add the attribute **na = " "**, it does not do anything. 

Appreciate your guidance!

**Comment by batpigandme** [comment]
@KHBHH,
This sort of question is a better fit for https://community.rstudio.com. Do you mind asking it over there? (You might want to read https://www.tidyverse.org/help/ first to maximise your chances of getting a good answer.)
